ArduPlane/altitude.cpp

```diff
@@ -315,8 +307,19 @@ void Plane::set_target_altitude_proportion(const Location &loc, float proportion
 {
     set_target_altitude_location(loc);
     proportion = constrain_float(proportion, 0.0f, 1.0f);
+
+    // We only do altitude slope handling when above CLMB_SLOPE_HGT or when
+    // descending. This is meant to prevent situations where we try to slowly
+    // gain height at low altitudes, potentially hitting obstacles.
+    if (target_altitude.offset_cm > 0 &&
+        adjusted_relative_altitude_cm() < g2.waypoint_climb_slope_height_min * 100) {
+        plane.set_target_altitude_location(plane.next_WP_loc);
```

Comment:
Should use the passed in location here. I also think its odd that your setting to the destination altitude here. That means that the altitude target will jump up to the destination, then when you climb above the min height it jumps back down to min height and then ramps from there. Better to constrain up to min height so that the altitude is constant before the ramp rather than having step changes. (you will also have to constrain down to the altitude of the passed in location).

Reply:
Hum, I guess the problem is using the true height as feedback. We could remove that and just shape the desired altitude.